api/jobs.go

```diff
@@ -158,52 +204,86 @@ func (api *API) unlockJob(ctx context.Context, lockID string) {
-			http.Error(w, "invalid path parameter - count should be a positive integer", http.StatusBadRequest)
-			return
-		}
+	numTasks, err := strconv.Atoi(count)
+	if err != nil {
+		log.Error(ctx, "invalid path parameter - failed to convert count to integer", err, logData)
+		http.Error(w, "invalid path parameter - failed to convert count to integer", http.StatusBadRequest)
+		return
+	}
 
-		lockID, err := api.dataStore.AcquireJobLock(ctx, id)
-		if err != nil {
-			log.Error(ctx, "acquiring lock for job ID failed", err, logData)
+	floatNumTasks := float64(numTasks)
+	isNegative := math.Signbit(floatNumTasks)
+	if isNegative {
+		err = errors.New("the count is negative")
+	}
+	if err != nil {
+		log.Error(ctx, "invalid path parameter - count should be a positive integer", err, logData)
+		http.Error(w, "invalid path parameter - count should be a positive integer", http.StatusBadRequest)
+		return
+	}
+
+	lockID, err := api.dataStore.AcquireJobLock(ctx, id)
+	if err != nil {
+		log.Error(ctx, "acquiring lock for job ID failed", err, logData)
+		http.Error(w, serverErrorMessage, http.StatusInternalServerError)
+		return
+	}
+	defer api.unlockJob(ctx, lockID)
+
+	err = api.dataStore.PutNumberOfTasks(req.Context(), id, numTasks)
+	if err != nil {
+		log.Error(ctx, "putting number of tasks failed", err, logData)
+		if err == mongo.ErrJobNotFound {
+			http.Error(w, "Failed to find job in job store", http.StatusNotFound)
+		} else {
 			http.Error(w, serverErrorMessage, http.StatusInternalServerError)
-			return
 		}
-		defer api.unlockJob(ctx, lockID)
+		return
+	}
 
-		err = api.dataStore.PutNumberOfTasks(req.Context(), id, numTasks)
-		if err != nil {
-			log.Error(ctx, "putting number of tasks failed", err, logData)
-			if err == mongo.ErrJobNotFound {
-				http.Error(w, "Failed to find job in job store", http.StatusNotFound)
-			} else {
-				http.Error(w, serverErrorMessage, http.StatusInternalServerError)
-			}
-			return
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+}
+
+// closeResponseBody closes the response body and logs an error if unsuccessful
+func closeResponseBody(ctx context.Context, resp *http.Response) {
+	if resp.Body != nil {
+		if err := resp.Body.Close(); err != nil {
+			log.Error(ctx, "error closing http response body", err)
 		}
+	}
+}
 
-		w.Header().Set("Content-Type", "application/json")
-		w.WriteHeader(http.StatusOK)
+// updateSearchIndexName calls the GetIndexNameFromResponse function, in the reindex package, to get the index name that was returned by the Search API.
+// It then calls the UpdateIndexName function, in the mongo package, to update the search_index_name value in the relevant Job Resource in the data store.
+func (api *API) updateSearchIndexName(ctx context.Context, reindexResponse *http.Response, searchErr error, newJob models.Job, id string) (models.Job, error) {
+	defer closeResponseBody(ctx, reindexResponse)
```

Comment:
is it better practice to defer closeResponseBody at the point where we know it was created (before calling this function?)

Reply:
maybe next time. If it's needed there is another logic branch where this isn't happening.

Reply:
ok no big deal